File: TransCoder_tokenized_test_set_functions/java/PROGRAM_CENSOR_WORD_ASTERISKS_SENTENCE.java
static String censor ( String text, String word ) {
  String [ ] word_list = text . split ( "\\s+" );
  String result = "";
  String stars = "";
  for ( int i = 0;
  i < word . length ( );
  i ++ ) stars += '*';
  int index = 0;
  for ( String i : word_list ) {
    if ( i . compareTo ( word ) == 0 ) word_list [ index ] = stars;
    index ++;
  }
  for ( String i : word_list ) result += i + ' ';
  return result;
}

File: TransCoder_tokenized_test_set_functions/cpp/PROGRAM_CENSOR_WORD_ASTERISKS_SENTENCE.cpp
string censor ( string text, string word ) {
  vector < string > word_list;
  boost :: split ( word_list, text, boost :: is_any_of ( "\\ +" ) );
  string result = "";
  string stars = "";
  for ( int i = 0;
  i < word . size ( );
  i ++ ) stars += '*';
  int index = 0;
  for ( string i : word_list ) {
    if ( i . compare ( word ) == 0 ) {
      word_list [ index ] = stars;
    }
    index ++;
  }
  for ( string i : word_list ) {
    result += i + ' ';
  }
  return result;
}